simulator_service/src/simulator_service_node.cpp

```diff
@@ -0,0 +1,91 @@
+#include <ros/ros.h>
+
+#include <simulator_service/SimulateTrajectory.h>
+
+#include <actionlib/client/simple_action_client.h>
+#include <control_msgs/FollowJointTrajectoryAction.h>
+
+static void scaleDurations(trajectory_msgs::JointTrajectory& traj, double scale)
+{
+  for (std::size_t i = 0; i < traj.points.size(); ++i)
+  {
+    traj.points[i].time_from_start = traj.points[i].time_from_start * scale;
+  }
+}
+
+namespace simulator_service
+{
+  class SimulatorService
+  {
+  public:
+    SimulatorService(ros::NodeHandle& nh, const std::string& service_name)
+      : ac_("joint_trajectory_action", true)
+      , service_name_(service_name)
+    {
+      execute_trajectory_service_ = nh.advertiseService(service_name_, 
+        &SimulatorService::simulateTrajectoryCallback, 
+        this);
+
+      // Establish connection to robot action server
+      if (!ac_.waitForServer(ros::Duration(5.0)))
+      {
+        ROS_ERROR_STREAM("Could not connect to action server");
+        throw std::runtime_error("Could not connect to joint_trajectory_action server");
```

Comment:
If the action server topic name is made configurable, this line should be changed as well.